cloud/scope/powervs_cluster_test.go

```diff
@@ -1234,20 +1321,51 @@ func TestReconcilePowerVSServiceInstance(t *testing.T) {
 		g.Expect(*clusterScope.IBMPowerVSCluster.Status.ServiceInstance.ControllerCreated).To(BeFalse())
 	})
 
-	t.Run("When create service instance return error", func(t *testing.T) {
+	t.Run("When service instance id is set in both spec and status, ID from status takes precedence", func(t *testing.T) {
 		g := NewWithT(t)
 		setup(t)
 		t.Cleanup(teardown)
 
-		clusterScopeParams := getPowerVSClusterScopeParams()
-		clusterScopeParams.ResourceControllerFactory = func() (resourcecontroller.ResourceController, error) {
-			mockResourceController.EXPECT().GetServiceInstance(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil)
-			mockResourceController.EXPECT().CreateResourceInstance(gomock.Any()).Return(nil, nil, errors.New("failed to create resource instance"))
-			return mockResourceController, nil
+		clusterScope := PowerVSClusterScope{
+			ResourceClient: mockResourceController,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					ServiceInstance: &infrav1beta2.IBMPowerVSResourceReference{
+						ID: ptr.To(serviceInstanceID),
+					},
```

Comment:
Id is same in both spec and status, It is supposed to be different for this case right?

Reply:
Yes, updated it